packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts

```diff
@@ -49,17 +48,57 @@ export function setS3UtilsInstance(s3Utils: S3Utils): void {
   s3UtilsInstance = s3Utils;
 }
 
+function documentResponseContainsMultipartCidReference(
+  documentResponse: DocumentResponse
+): documentResponse is DocumentResponse & { Document: { Include: { _href: string } } } {
+  return (
+    typeof documentResponse.Document !== "string" && !!documentResponse.Document?.Include?._href
+  );
+}
+
+function documentResponseContainsDocument(
+  documentResponse: DocumentResponse
+): documentResponse is DocumentResponse & { Document: string } {
+  return typeof documentResponse.Document === "string";
+}
```

Comment:
nit: Both return true if the doc resp contains Document (and additional checks), maybe add the specifics on the name as well? Like:
- `documentResponseContainsMultipartCidReferenceDocument`
- `documentResponseContainsStringDocument`

Reply:
I changed to be more specific:

- `documentResponseContainsMultipartCidReferenceToDocument`
- `documentResponseContainsDocumentInSoapMessage`